### ISSUE TITLE
[FG 2.1-snapshot] Zombie classes in forgeSrc.jar

### ISSUE THREAD

**Issue by Glassmaker** [MinecraftForge/ForgeGradle#303]
I have found that upgrading forge versions in the MDK project for example from 1589 to 1592 doesn't clean the project's temp folder "project_home/build/tmp" before recompiling to forgeSrc jar 

This leads to creating a hybrid forgeSrc jar including deleted classes and interfaces like IItemRenderer and GameData's inner class GameDataSnapshot as shown in the screenshot below

![fg-bug](https://cloud.githubusercontent.com/assets/5352172/11453554/dba81744-9620-11e5-953b-c4a73bcfb3e3.png)

These classes only appear in forgeSrc binary jar, forgeSrc-source.jar doesn't include them.

**Steps to reproduce:**
1- Make sure that you don't have "1.8.8-11.14.4.1589-1.8.8" and "1.8.8-11.15.0.1592-1.8.8"  in the cache folder "~/.gradle/caches/minecraft/net/minecraftforge/forge"
2- Download forge's 11.14.4.1589 MDK from http://files.minecraftforge.net/
3- Extract the zip file, and run gradlew setupDecompWorkspace.
4- Update the version in gradle.build file to "1.8.8-11.15.0.1592-1.8.8"
5- Run gradlew setupDecompWorkspace.
6- check "1.8.8-11.15.0.1592-1.8.8/snapshot/20151122" in the cache folder for the hybrid forgeSrc jar

**Workaround:**
1- Delete "1.8.8-11.15.0.1592-1.8.8" folder in the cache folder.
2- Delete the project's temp folder "project_home/build/tmp" to ensure a clean source code was generated prior to recompiling.
3- Run gradlew setupDecompWorkspace.

ForgeGradle version: ForgeGradle 2.1-SNAPSHOT-8971088

**Comment by cpw** [comment]
Gradle clean doesn't work?

**Comment by Glassmaker** [comment]
Before Step 5, yes.
After Step 5, no, recompiling is skipped.

**Comment by AbrarSyed** [comment]
has also been merged into FG 2.1
